### ISSUE TITLE
[FR] Run BedrockServer and EducationServer on the same machine

### ISSUE THREAD

**Issue by HandyHat** [bundabrg/GeyserReversion#30]
This is quite a big ask, and might not actually be feasible, but it would be quite cool if we could have Bedrock and Education players able to connect to the same instance of Geyser - I assume the servers would have to run on different ports, and perhaps 2 versions of Reversion would have to run in parallel?

**Comment by bundabrg** [comment]
Its definitely do-able (to a degree) and on my roadmap possibly as the next feature.

I can't run both on the same port but GeyserReversion will provide in the config file the ability to listen on other ports and each port can be configured as to whether it is Education or Bedrock. A possibly future feature (if its of any use, it may not have a use) is to also limit what versions are allowed on what port etc.

**Comment by gjorgdy** [comment]
Would it be a possibility to just run an Education Edition port and let Bedrock be, as I don't need version support for Bedrock. I only need players on Education Edition to be able to join.

**Comment by bundabrg** [comment]
Yes. You install this plugin, and in its config set the edition to 'education'. Thats how it works right now in that you have to choose either/or. When I ever get around to allowing both at the same time you would just set what edition on what ports (since both can't run on the same port) and can in this case just enable edition as well.

Let me know if you have any issues with Ed (that effect ed users in particular).

**Comment by bundabrg** [comment]
As of now, GeyserReversion can't run both at the same time because Bedrock and Education both have identical looking 'connection' attempts but if they get the wrong response they will disconnect with an error. So we have to run them on 2 different ports.

My plan is to eventually allow GeyserReversion to support listening on multiple ports and you can then assign what edition (and perhaps even what versions) will work on each port. This will help those who run only the plugin version of Geyser.

Till then my recommendation is:

  * Run 2 Geysers. One is configured for Bedrock and the other configured for Education. Both can be standalone or one can be standalone and the other a plugin version of Geyser.
  * They will need to listen on different ports to each other
  * They will both connect to the same java server

I personally run 2 standalone versions at the moment.

**Comment by WolfyRed** [comment]
> 
> 
> As of now, GeyserReversion can't run both at the same time because Bedrock and Education both have identical looking 'connection' attempts but if they get the wrong response they will disconnect with an error. So we have to run them on 2 different ports.
> 
> My plan is to eventually allow GeyserReversion to support listening on multiple ports and you can then assign what edition (and perhaps even what versions) will work on each port. This will help those who run only the plugin version of Geyser.
> 
> Till then my recommendation is:
> 
>     * Run 2 Geysers. One is configured for Bedrock and the other configured for Education. Both can be standalone or one can be standalone and the other a plugin version of Geyser.
> 
>     * They will need to listen on different ports to each other
> 
>     * They will both connect to the same java server
> 
> 
> I personally run 2 standalone versions at the moment.

Another question, will you still support 1.16.x after 1.17 releases? I'm asking because some players on my server play on iOS 9.0 and they will be stuck on 1.16.x due to that.
